enclave/core/switchlesscalls.c

```diff
@@ -25,7 +28,13 @@ static oe_host_worker_context_t* _host_worker_contexts = NULL;
 */
 bool oe_is_switchless_initialized()
 {
-    return _host_worker_count != 0;
+    bool is_initialized;
+
+    oe_spin_lock(&_lock);
+    is_initialized = _host_worker_count != 0;
```

Comment:
Suggestion: Consider using acquire-release on the _host_worker_count instead of a spinlock in this function.